core/src/main/java/nu/yona/server/crypto/seckey/DateFieldEncryptor.java

```diff
@@ -0,0 +1,44 @@
+/*******************************************************************************
+ * Copyright (c) 2015, 2017 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.crypto.seckey;
+
+import java.time.LocalDate;
+import java.util.Base64;
+
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+
+@Converter
+public class DateFieldEncryptor implements AttributeConverter<LocalDate, String>
+{
+	@Override
+	public String convertToDatabaseColumn(LocalDate attribute)
+	{
+		String retVal = (attribute == null) ? null : Base64.getEncoder().encodeToString(SecretKeyUtil.encryptDate(attribute));
+		if (retVal == null)
+		{
+			throw new IllegalStateException("YD-441: lastStatusChangeTime cannot be null. Input value: " + attribute);
```

Comment:
This exception message seems to be wrong...

Reply:
This is another copy/paste error. Sorry about having so many.
I've removed this check and replaced it here and in ``DateTimeFieldEncryptor`` with an assert.